README.md

```diff
@@ -1,101 +1,224 @@
+## Index
+* [Supported platforms](#Supported-platforms)
+* [Prerequisites and dependencies](#Prerequisites-and-dependencies)
+* [Building the project](#Building-the-project)
+* [Running the project](#Running-the-project)
+* [Making use of the mounted environment](#Making-use-of-the-mounted-environment)
+* [Common issues and troubleshooting](#Common-issues-and-troubleshooting)
 
-## Prerequisite dependencies
-1. Java
-2. Maven
-3. libfuse for Linux / osxfuse for macOS
+## Supported platforms
+* Linux (Debian/Fedora)                                                         
+* macOS (via [osxfuse](https://osxfuse.github.io/))
+* Windows (via Windows Subsystem for Linux version 2 - [WSL2](https://codefellows.github.io/setup-guide/windows/))
 
-#### Linux
+## Prerequisites and dependencies
+This tool was programmed in Java, and depends on the following software:
 
-[`libfuse`](https://github.com/libfuse/libfuse) needs to be installed.
+| Software        | Tested version  | Operating system  |
+| -------------: |:-------------|:-------------|
+| [Java](https://www.java.com/) | OpenJDK 11.0.10 | Linux and macOS |
+| [Maven](https://maven.apache.org/)  | Apache Maven 3.6.3 | Linux and macOS |
+| [libfuse](https://github.com/libfuse/libfuse) | libfuse-dev 2.9.9-3 | Linux |
+| [osxfuse](https://osxfuse.github.io/) | 3.11.2 | macOS |
 
-##### Ubuntu
-```
-sudo apt-get install libfuse-dev
-``` 
+Column "Tested version" represents each software's version used on our latest tests, serving as a reference. Nevertheless, we encourage you to download their current version and check its functionality.
 
-##### RedHat
-```
-yum install fuse fuse-devel
-``` 
+### Installing dependencies
 
-#### macOS
+Herebelow you can find commands to **check** if the dependencies have already been installed, and **how to install** them if not. For each of them, the first command should prompt the version of the installed dependency. If not installed, run its second command to install it. 
 
-[`osxfuse`](https://osxfuse.github.io) needs to be installed.
+#### 1. Linux
+Depending on your Linux distribution (Ubuntu or RedHat) use its corresponding commands.
 
+#### 1.1 Ubuntu
+Run the following command before any installation.
+``` bash
+sudo apt update
 ```
-brew cask install osxfuse
-```
+Checking installation and installing Java, Maven and libfuse (respectively).
+``` bash
+# Java 
+java -version
+sudo apt install default-jdk
 
+# Maven
+mvn -v
+sudo apt-get install maven
 
-## Build the project
-First, clone the project to your local machine. Then go inside the cloned directory and build the project using the command below. It will produce the executable jar file in the /target directory.
+# Libfuse
+fusermount -V
+sudo apt-get install libfuse-dev
 ```
-mvn clean install
+#### 1.2 RedHat
+Run the following command before any installation.
+``` bash
+yum update
 ```
+Checking installation and installing Java, Maven and libfuse (respectively).
+``` bash
+# Java 
+#! ? 
+#! ?
+
```

Comment:
For RedHat, java & maven installation commands refer here https://tecadmin.net/install-apache-maven-on-centos/